src/scripts/main.js

```diff
@@ -1,3 +1,38 @@
 'use strict';
 
-// write code here
+const tableHeader = document.querySelector('tr');
+const tableBody = document.querySelector('tbody');
+const toNumber = (string) => {
+  if (typeof string === 'string') {
+    const clearNum = string.replace(/[$, ]/g, '');
+
+    return parseInt(clearNum);
+  }
+};
+
+tableHeader.addEventListener('click', (e) => {
+  const headerChildren = [...tableHeader.children];
+  const columnName = e.target.closest('th');
+  const sortedColumn = headerChildren.indexOf(columnName);
+  const columnsToSort = [...tableBody.children];
+
+  const columnsSorted = columnsToSort.sort((a, b) => {
+    const elemA = a.children[sortedColumn].innerHTML;
+    const elemb = b.children[sortedColumn].innerHTML;
+
+    const aToNum = toNumber(elemA);
+    const bToNum = toNumber(elemb);
+
+    if (isNaN(aToNum)) {
```

Comment:
You're using 'isNaN' to check if the conversion to a number failed, but 'isNaN' will also return true for values like 'Infinity'. Consider using 'Number.isNaN' for a more precise check.